_episodes_rmd/07-control-flow.Rmd

```diff
@@ -430,14 +430,14 @@ output_vector2
 > > ```
 > >
 > > We will now store these countries in a variable called candidateCountries, and then loop over each entry in the variable.
-> > Inside the loop, we evaluate the average life expectancy for each country, and if the average life expectancy is less than 50 we use base-plot to plot the evolution of average life expectancy:
+> > Inside the loop, we evaluate the average life expectancy for each country, and if the average life expectancy is less than 50 we use base-plot to plot the evolution of average life expectancy using `with` and `subset`:
 > >
 > > ```{r 07-chall-05-sol-c, eval=FALSE}
 > > thresholdValue <- 50
 > > candidateCountries <- grep("^B", unique(gapminder$country), value=TRUE)
 > >
 > > for( iCountry in candidateCountries){
-> >     tmp <- mean(subset(gapminder, country==iCountry)$lifeExp)
+> >     tmp <- mean(gapminder[gapminder$continent == iContinent, "lifeExp"])
```

Comment:
Isn't this going to be a problem because the loop is iterating over countries and not continents?